pkg/ui/workspaces/e2e-tests/build/start-crdb-movr.sh

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+print_usage() {
+  cat<< EOF
+USAGE: $0 COMMAND [ARGS...]
+
+Starts a CockroachDB 'movr' demo, executing COMMAND with ARGS when the database
+serves HTTP traffic on port 8080. The path to a file containing the connection
+URL (from which a username and password can be extracted with standard URL
+parsing) is provided to COMMAND via the \$CONN_URL_FILE environment variable.
+
+When COMMAND exits, the database is stopped.
+
+EXAMPLES:
+    $0 'curl http://localhost:8080'
+        Load index.html once the database has started.
+
+    $0 'cat \$CONN_URL_FILE'
+        Print the connection string once the database has started.
+EOF
+}
+
+EXIT_CODE=-1
+cleanup() {
+  # Send an exit signal to the background CRDB job.
+  kill %1
+
+  # Wait for the background CRDB job to exit, using 'set +e' to allow 'wait' to
+  # exit with non-zero. CRDB will exit non-zero (and 'wait' will too), but with
+  # 'set -e' this entire script would normally exit *immediately*.
+  set +e; wait %1; set -e
+
+  # Forward the exit code from COMMAND to the calling shell.
+  exit $EXIT_CODE;
+}
+
+# Command used improperly; print help and exit non-zero.
+if [ $# -lt 1 ] || [ -z "$1" ]; then
+  print_usage
+  exit 1
+fi
+
+# Help explicitly requested; print help and exit zero.
+if [ "$1" == "help" ] || [ "$1" == "-h" ] || [ "$1" == "--help" ]; then
+  print_usage
+  exit 0
+fi
+
+if curl --silent --head http://localhost:8080 > /dev/null; then
+  cat<< EOF
+ERROR: Another process is already listening on port 8080.  It may be a CRDB
+       instance, but to be safe, this script is exiting early.
+
+       Here's what we know about that process, courtesy of 'ps':\n
+EOF
+
+  ps -o pid -o command -p $(lsof -t -i :8080)
+  exit 2
+fi
+
+# Accept an arbitrary Cockroach binary via the $COCKROACH environment variable
+CRDB=${COCKROACH:-../../../../cockroach}
+
+# Start a 'movr' demo cluster, writing the connection URL to a file.
+export CONN_URL_FILE=$(mktemp)
+$CRDB demo movr \
+  --nodes=1 \
+  --multitenant=false \
+  --http-port=8080 \
+  --listening-url-file=$CONN_URL_FILE \
+  --execute 'select pg_sleep(1000000)' &
```

Comment:
Curious why `movr`? Isn't intended for interactive sessions? Why not `start-singlenode --insecure`?

Reply:
`start-single-node` required a significant amount of setup before the tests could start executing — mostly generating certs, creating a non-root user with a password, creating a few tables, etc.  `cockroach workload` helps with that slightly, but doing it manually winds up being quite a bit slower than `cockroach demo movr` as well.  I found an early attempt at that and it was slower and more effort for no benefit :) https://gist.github.com/sjbarag/30eb3c3e253235fef00c461affa1348f

Curiously, a lot of that is handled in [build/deploy/cockroach.sh](https://github.com/cockroachdb/cockroach/blob/e52f732f8130df956449786c053d1d4539548411/build/deploy/cockroach.sh) (the entrypoint for the `cockroachdb/cockroach` and `cockroachdb/cockroach-ci` docker images).  I've got a WIP implementation that uses `docker compose` for these purposes in CI, so I'll see if I can make that script work for dev uses as well.

Reply:
>Curiously, a lot of that is handled in [build/deploy/cockroach.sh](https://github.com/cockroachdb/cockroach/blob/e52f732f8130df956449786c053d1d4539548411/build/deploy/cockroach.sh)

Right, `democluster.NewDemoCluster` does essentially the same bootstrapping as in `cockroach.sh` but then switches into `ShellCtx`. It looks like this predates `cockroach.sh` [1] which was added relatively recently and expressly for use with Docker :)

[1] https://github.com/cockroachdb/cockroach/commit/2862374a4743aece5707ea52d0ac17f6cb10dc7b

> I've got a WIP implementation that uses docker compose for these purposes in CI, so I'll see if I can make that script work for dev uses as well.

I think that would be nicer, thanks!